fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -506,6 +506,26 @@ ACTOR Future<Void> commitBatch(
 	state Optional<Key> lockedKey = self->txnStateStore->readValue(databaseLockedKey).get();
 	state bool locked = lockedKey.present() && lockedKey.get().size();
 
+	state Optional<Key> onlySystemKey = self->txnStateStore->readValue(onlySystemTransactionsKey).get();
+	state bool onlySystem = onlySystemKey.present() && onlySystemKey.get().size();
+
+	if(onlySystem) {
+		for (int t = 0; t<trs.size(); t++) {
+			if( committed[t] == ConflictBatch::TransactionCommitted ) {
+				bool foundSystem = false;
+				for(auto& m : trs[t].transaction.mutations) {
+					if( ( m.type == MutationRef::ClearRange ? m.param2 : m.param1 ) >= nonMetadataSystemKeys.end) {
+						foundSystem = true;
```

Comment:
Does this allow a transaction that modifies non-system keys if it also modifies system keys?

Reply:
yes, I figured that was the safer approach

Reply:
the only candidate transactions that fit into the mixed category are probably ones that lock the database

Reply:
I think this is potentially confusing, as the name of the key could lead someone to believe that writes outside the system keys are not possible, but as a feature that may be a reasonable choice.